src/main/java/io/vertx/core/net/impl/pool/CombinerExecutor.java

```diff
@@ -27,7 +28,12 @@ public class CombinerExecutor<S> implements Executor<S> {
   private final Queue<Action<S>> q = PlatformDependent.newMpscQueue();
   private final AtomicInteger s = new AtomicInteger();
   private final S state;
-  private final ThreadLocal<Task> current = new ThreadLocal<>();
+
+  protected final class InProgressTail {
+    Task task;
+  }
+
+  private final FastThreadLocal<InProgressTail> current = new FastThreadLocal<>();
```

Comment:
note this will be a problem when we will use Netty 5 and Netty 4

Reply:
It will be something to be review, but won't break, because it is still backed by JDK thread local in case the executing thread isn't a fast thread local.

Reply:
my point is that, this will not be the same Java class in Netty 4 / Netty 5

Reply:
Yep, that's a good point indeed...I have a solution that doesn't use *any* thread locals, but it would complicate a lot the code there, really

Reply:
yeah but we would need that eventually I think, so perhaps it is best avoiding thread local if we can now that we have the time to do it ?

Reply:
I can prepare the a different branch with another version but will use atomic queues to achieve the same, probably with worse performance.
Another version could change a lot the API and pass a queue of existing post actions to let the actions to enqueue themselves but it will super invasive and complex imo.
In the current form this one already deliver what we need Ie avoid ThreadLocal::remove and GC nepotism, so I would prefer to do it in 2 separate steps

Reply:
fine